bin/deliver_reviewed_data.py

```diff
@@ -352,27 +405,77 @@ def deliver_data(self, project_id=None, sample_id=None):
             for project in project_to_samples:
                 # Create the batch directory
                 today = datetime.date.today().isoformat()
-                batch_delivery_folder = os.path.join(delivery_dest, project, today)
+                batch_delivery_folder = os.path.join(self.delivery_dest, project, today)
                 os.makedirs(batch_delivery_folder, exist_ok=True)
-                # move all the staged sample directory
+                # Move all the staged sample directory
                 project_to_delivery_folder[project] = batch_delivery_folder
                 for sample in project_to_samples.get(project):
-                    shutil.move(sample2stagedirectory.get(sample.get(ELEMENT_SAMPLE_INTERNAL_ID)),
-                                batch_delivery_folder)
+                    shutil.move(
+                        sample2stagedirectory.get(sample.get(ELEMENT_SAMPLE_INTERNAL_ID)),
+                        batch_delivery_folder
+                    )
+                    self.update_registered_files(sample.get(ELEMENT_SAMPLE_INTERNAL_ID), batch_delivery_folder)
                 self.write_metrics_file(project, batch_delivery_folder)
                 self.generate_md5_summary(project, batch_delivery_folder)
-
             self.mark_samples_as_released(list(sample2stagedirectory))
 
         self.cleanup()
-        # TODO: Generate project report
+
+        # Generate project report
+        project_to_reports = {}
+        for project in project_to_samples:
+            project_report = join(self.delivery_dest, project, 'project_%s_report.pdf' % project)
+            try:
+                pr = ProjectReport(project)
+                pr.generate_report('pdf')
+            except Exception:
+                self.critical('Project report generation for %s failed' % project)
+                project_report = None
+            if project_report:
+                project_to_reports[project] = project_report
+
+        # Send email confirmation with attachments
+        self.email_report(project_to_samples, project_to_reports)
+
+    def email_report(self, project_to_samples, project_to_reports):
+        msg = self.create_email_report(project_to_samples)
+        project_ids = ', '.join(list(project_to_samples))
+        self.info(msg)
+        if self.email:
+            send_email(
+                msg=msg,
+                subject='Data delivery: ' + project_ids,
+                attachments=project_to_reports.values(),
```

Comment:
The attachments will only work with the newest (unreleased) version of EGCG-Core